File: Assets/Script/Lib/Agent/Tools.ts
// @preview-file off clear
export { planTruncatedEditRecovery } from 'Agent/Tool/TruncatedEditRecovery';
export type { TruncatedEditRecoveryNotice, TruncatedEditRecoveryPlan } from 'Agent/Tool/TruncatedEditRecovery';
export { sendWebIDEFileUpdate, sendWebIDERefreshTree } from 'Agent/Tool/WebIDESync';
export {
	getLogs,
	listFiles,
	readFile,
	readFileRaw,
	inspectWorkspaceTextTarget,
	searchFiles,
} from 'Agent/Tool/Workspace';
export type {
	ListFilesResult,
	GetLogsResult,
	ReadFileResult,
	WorkspaceTextTargetResult,
	SearchFilesToolResult,
} from 'Agent/Tool/Workspace';

export { searchDoraDoc, searchDoraDocHttp, readDoraDoc } from 'Agent/Tool/DoraDocSearch';
export type {
	DoraDocLanguage,
	DoraDocSearchType,
	DoraDocProgrammingLanguage,
	DoraDocSearchHit,
	DoraDocSearchResult,
	DoraDocReadResult,
} from 'Agent/Tool/DoraDocSearch';


export {
	createTask,
	setTaskStatus,
	listCheckpointsForTasks,
	listCheckpoints,
	getCheckpoint,
	summarizeTaskChangeSet,
	getTaskChangeSetDiff,
	applyFileChanges,
	deleteFile,
	rollbackCheckpoint,
	rollbackTaskChangeSet,
	getCheckpointEntriesForDebug,
	getCheckpointDiff,
} from 'Agent/Tool/Checkpoint';
export type {
	AgentTaskStatus,
	AgentTaskWorkMode,
	CheckpointStatus,
	FileOp,
	FileChange,
	ApplyChangesOptions,
	CreateTaskResult,
	ApplyChangesResult,
	DeleteFileResult,
	RollbackResult,
	TaskRollbackResult,
	CheckpointDiffFile,
	CheckpointDiffResult,
	CheckpointItem,
	TaskChangeSetFile,
	TaskChangeSetSummary,
} from 'Agent/Tool/Checkpoint';

export { build, runSingleTsTranspile } from 'Agent/Tool/Build';
export type { BuildMessage, BuildResult } from 'Agent/Tool/Build';

export { fetchUrl } from 'Agent/Tool/Fetch';
export type { FetchUrlMode, FetchUrlProgress, FetchUrlResult } from 'Agent/Tool/Fetch';

export { executeCommand } from 'Agent/Tool/Command';
export type {
	ExecuteCommandMode,
	ExecuteCommandProgress,
	ExecuteCommandResult,
} from 'Agent/Tool/Command';

File: Assets/Script/Lib/Agent/Tools.lua
-- [ts]: Tools.ts
local ____exports = {} -- 1
do -- 1
	local ____TruncatedEditRecovery = require("Agent.Tool.TruncatedEditRecovery") -- 2
	____exports.planTruncatedEditRecovery = ____TruncatedEditRecovery.planTruncatedEditRecovery -- 2
end -- 2
do -- 2
	local ____WebIDESync = require("Agent.Tool.WebIDESync") -- 4
	____exports.sendWebIDEFileUpdate = ____WebIDESync.sendWebIDEFileUpdate -- 4
	____exports.sendWebIDERefreshTree = ____WebIDESync.sendWebIDERefreshTree -- 4
end -- 4
do -- 4
	local ____Workspace = require("Agent.Tool.Workspace") -- 5
	____exports.getLogs = ____Workspace.getLogs -- 6
	____exports.listFiles = ____Workspace.listFiles -- 7
	____exports.readFile = ____Workspace.readFile -- 8
	____exports.readFileRaw = ____Workspace.readFileRaw -- 9
	____exports.inspectWorkspaceTextTarget = ____Workspace.inspectWorkspaceTextTarget -- 10
	____exports.searchFiles = ____Workspace.searchFiles -- 11
end -- 11
do -- 11
	local ____DoraDocSearch = require("Agent.Tool.DoraDocSearch") -- 21
	____exports.searchDoraDoc = ____DoraDocSearch.searchDoraDoc -- 21
	____exports.searchDoraDocHttp = ____DoraDocSearch.searchDoraDocHttp -- 21
	____exports.readDoraDoc = ____DoraDocSearch.readDoraDoc -- 21
end -- 21
do -- 21
	local ____Checkpoint = require("Agent.Tool.Checkpoint") -- 32
	____exports.createTask = ____Checkpoint.createTask -- 33
	____exports.setTaskStatus = ____Checkpoint.setTaskStatus -- 34
	____exports.listCheckpointsForTasks = ____Checkpoint.listCheckpointsForTasks -- 35
	____exports.listCheckpoints = ____Checkpoint.listCheckpoints -- 36
	____exports.getCheckpoint = ____Checkpoint.getCheckpoint -- 37
	____exports.summarizeTaskChangeSet = ____Checkpoint.summarizeTaskChangeSet -- 38
	____exports.getTaskChangeSetDiff = ____Checkpoint.getTaskChangeSetDiff -- 39
	____exports.applyFileChanges = ____Checkpoint.applyFileChanges -- 40
	____exports.deleteFile = ____Checkpoint.deleteFile -- 41
	____exports.rollbackCheckpoint = ____Checkpoint.rollbackCheckpoint -- 42
	____exports.rollbackTaskChangeSet = ____Checkpoint.rollbackTaskChangeSet -- 43
	____exports.getCheckpointEntriesForDebug = ____Checkpoint.getCheckpointEntriesForDebug -- 44
	____exports.getCheckpointDiff = ____Checkpoint.getCheckpointDiff -- 45
end -- 45
do -- 45
	local ____Build = require("Agent.Tool.Build") -- 66
	____exports.build = ____Build.build -- 66
	____exports.runSingleTsTranspile = ____Build.runSingleTsTranspile -- 66
end -- 66
do -- 66
	local ____Fetch = require("Agent.Tool.Fetch") -- 69
	____exports.fetchUrl = ____Fetch.fetchUrl -- 69
end -- 69
do -- 69
	local ____Command = require("Agent.Tool.Command") -- 72
	____exports.executeCommand = ____Command.executeCommand -- 72
end -- 72
return ____exports -- 72